src/ansi-c/c_typecheck_base.cpp

```diff
@@ -748,11 +748,15 @@ void c_typecheck_baset::typecheck_declaration(
       typet ret_type = void_type();
       if(new_symbol.type.id()==ID_code)
         ret_type=to_code_type(new_symbol.type).return_type();
-      assert(parameter_map.empty());
       if(ret_type.id()!=ID_empty)
+      {
+        DATA_INVARIANT(
```

Comment:
If `parameter_map` should always be empty then `DATA_INVARIANT`.  If it should be empty at this point then `INVARIANT`.